src/metaswitch/clearwater/etcd_shared/common_etcd_synchronizer.py

```diff
@@ -171,11 +172,16 @@ def read_from_etcd(self, wait=True):
             if wait:
                 # If the cluster view hasn't changed since we last saw it, then
                 # wait for it to change before doing anything else.
-                _log.info("Read value {} from etcd, "
-                          "comparing to last value {}".format(
-                              result.value,
-                              self._last_value))
-
+                # We need to pass the config value to a function which can
+                # prepare it for logging by encoding it as utf-8.
+                if self._last_value:
```

Comment:
Why have you changed the if here? Did you hit issues with the logging function if you passed in an empty value?